File: Sorts/pyalgoritmos/list100.py
import random
def selectionSort(array):

    size = len(array)
    
    for ind in range(size):
        
        min_index = ind
 
        for j in range(ind + 1, size):
            
            # select the minimum element in every iteration
            if array[j] < array[min_index]:
                
                min_index = j
        
        # swapping the elements to sort the array
        (array[ind], array[min_index]) = (array[min_index], array[ind])

def bubbleSort(arr):
    
    n = len(arr)
    
    # optimize code, so if the array is already sorted, it doesn't need
    # to go through the entire process
    swapped = False
    
    # Traverse through all array elements
    for i in range(n-1):
        
        # range(n) also work but outer loop will
        # repeat one time more than needed.
        # Last i elements are already in place
        for j in range(0, n-i-1):
 
            # traverse the array from 0 to n-i-1
            # Swap if the element found is greater
            # than the next element
            if arr[j] > arr[j + 1]:
                
                swapped = True
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
         
        if not swapped:
            # if we haven't needed to make a single swap, we 
            # can just exit the main loop.
            return

def insertionSort(arr):
 
    # Traverse through 1 to len(arr)
    for i in range(1, len(arr)):
 
        key = arr[i]
 
        # Move elements of arr[0..i-1], that are
        # greater than key, to one position ahead
        # of their current position
        j = i-1
        
        while j >= 0 and key < arr[j] :
                
                arr[j + 1] = arr[j]
                j -= 1
        
        arr[j + 1] = key

list100py = []

while(len(list100py) < 100):
    
    num = random.randint(0, 100)

    list100py.append(num)

print(selectionSort(list100py))

File: Sorts/javaalgoritmos/list100.java
import java.util.*;
import java.io.*;
import java.lang.*;


public class list100 {
    
    public static void selectionSort(int[] arr){  
        
        for (int i = 0; i < arr.length - 1; i++){  
            
            int index = i;  
            
            for (int j = i + 1; j < arr.length; j++){  
                
                if (arr[j] < arr[index]){  
                    
                    index = j;//searching for lowest index  
                
                }  
            
            }  
            
            int smallerNumber = arr[index];   
            arr[index] = arr[i];  
            arr[i] = smallerNumber;  
        
        }  
    
    }
    static void bubbleSort(int[] arr) {  
        
        int n = arr.length;  
        int temp = 0;  
        
        for(int i=0; i < n; i++){  
                
            for(int j=1; j < (n-i); j++){  
                if(arr[j-1] > arr[j]){

                    //swap elements  
                    temp = arr[j-1];  
                    arr[j-1] = arr[j];  
                    arr[j] = temp;  
                }  
                          
            }  
        }
    }
    public static void insertionSort(int arr[]){
        
        int n = arr.length;
        
        for (int i = 1; i < n; ++i) {
            
            int key = arr[i];
            int j = i - 1;
 
            /* Move elements of arr[0..i-1], that are
               greater than key, to one position ahead
               of their current position */
            
           while (j >= 0 && arr[j] > key) {
                
                arr[j + 1] = arr[j];
                j = j - 1;
            
            }
            
            arr[j + 1] = key;
        
        }
    
    }  
    public static void main(String[] args) 
    { 
        
        int min = 1;
        int max = 101;
        
        int list100java[] = new int[101];

        for(int i = 0; i < 101; i++){

            Random random = new Random();
            int num = random.nextInt(max - min) + min;

            list100java[i] = num;

        }

        int new_list100java[] = new int[101];
        new_list100java = list100java;
        insertionSort(new_list100java);

        // for(int i = 0; i < 101; i++){

        //     System.out.printf("%d ", list100java[i]);

        // }
        // System.out.println("over");

        for(int i = 0; i < 101; i++){

            System.out.printf("%d ", new_list100java[i]);

        }

    }
    
}